src/index.js

```diff
@@ -1,110 +1,145 @@
-  //
-  //   [[plugins]]
-  //   package = "netlify-plugin-reflect"
-  //     [plugins.inputs]
-  //     foo = "bar"
-  inputs,
-  // `onError` event handlers receive the error instance as argument
-  error,
-
-  // Build constants
-  constants: {
-    // Path to the Netlify configuration file. `undefined` if none was used
-    CONFIG_PATH,
-    // Directory that contains the deploy-ready HTML files and assets
-    // generated by the build. Its value is always defined, but the target
-    // might not have been created yet.
-    PUBLISH_DIR,
-    // The directory where function source code lives.
-    // `undefined` if not specified by the user.
-    FUNCTIONS_SRC,
-    // The directory where built serverless functions are placed before
-    // deployment. Its value is always defined, but the target might not have
-    // been created yet.
-    FUNCTIONS_DIST,
-    // Boolean indicating whether the build was run locally (Netlify CLI) or
-    // in the production CI
-    IS_LOCAL,
-    // Version of Netlify Build as a `major.minor.patch` string
-    NETLIFY_BUILD_VERSION,
-    // The Netlify Site ID
-    SITE_ID,
-  },
-
-  // Core utilities
+export const onSuccess = async function ({
   utils: {
-    // Utility to report errors.
-    // See https://github.com/netlify/build#error-reporting
-    build,
-    // Utility to display information in the deploy summary.
-    // See https://github.com/netlify/build#logging
-    status,
-    // Utility for caching files.
-    // See https://github.com/netlify/build/blob/master/packages/cache-utils#readme
-    cache,
-    // Utility for running commands.
-    // See https://github.com/netlify/build/blob/master/packages/run-utils#readme
-    run,
-    // Utility for dealing with modified, created, deleted files since a git commit.
-    // See https://github.com/netlify/build/blob/master/packages/git-utils#readme
-    git,
-    // Utility for handling Netlify Functions.
-    // See https://github.com/netlify/build/tree/master/packages/functions-utils#readme
-    functions,
+    build, // Utility to report errors.
+    status, // Utility to display information in the deploy summary.
   },
 }) {
-  try {
-    // Commands are printed in Netlify logs
-    await run('echo', ['Hello world!\n'])
-  } catch (error) {
-    // Report a user error
-    build.failBuild('Error message', { error })
+  const {
+    REFLECT_API_KEY,
+    REFLECT_SUITE_ID,
+    REFLECT_WAIT_FOR_TEST_RESULTS,
+    REFLECT_DEFAULT_HOSTNAME,
+    PULL_REQUEST,
+    COMMIT_REF,
+    REPOSITORY_URL,
+    DEPLOY_PRIME_URL,
+  } = process.env;
+
+  if (!REFLECT_API_KEY) {
+    return build.failPlugin("REFLECT_API_KEY is not defined.");
   }
 
-  // Console logs are shown in Netlify logs
-  console.log('Netlify configuration', netlifyConfig)
-  console.log('Plugin configuration', inputs)
-  console.log('Build directory', PUBLISH_DIR)
+  if (!REFLECT_SUITE_ID) {
+    return build.failPlugin(
+      "suite_id must be defined in Plugin configuration."
```

Comment:
Just confirming this error should not have the REFLECT_ prefix

Reply:
Good catch - I think originally I had this as a field in the config file, but Netlify's docs suggest making all config values environmental variables so they can be configured in the UI. Will update this message.